spec/synchronization.lua

```diff
@@ -0,0 +1,78 @@
+describe("Synchronization", function()
+  local synchronization
+  local key = "foo"
+
+  before_each(function()
+    synchronization = require('resty.synchronization').new(1)
+  end)
+
+  it("Valid run execution", function()
+    local callback = spy.new(function() return 1 end)
+
+    local ret, result = synchronization:run(key, 10, callback)
+    assert.same(ret, true)
+    assert.same(result, {1})
+    assert.spy(callback).was.called()
```

Comment:
o we need the spies? I think that checking the results of `run` might be enough.

Reply:
It's just to make the test completed and easy to understand if someone review the code in future.

Reply:
:+1: